### PR TITLE
move ad-hoc attachment api calls to m365/api

### PR DESCRIPTION
**Pull request by ryanfkeepers**
finishes moving ad-hoc exchange api client calls from connector/exchange into m365/api.  Final cleanup for polish to follow.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #1996

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/exchange/restore_test.go [hunk 1]
```diff
@@ -138,6 +138,7 @@ func (suite *ExchangeRestoreIntgSuite) TestRestoreEvent() {
 				ctx,
 				test.bytes,
 				suite.ac.Events(),
+				suite.ac.Events(),
```
Comment (ashmrtn) [inline src/internal/connector/exchange/restore_test.go:141]: maybe we should make another interface that is something like `itemAndAttachmentPoster` and just pass in a single instance of the api layer. I'm not really a fan of passing in/having multiple references to the same underlying object. It seems like a good way run into unforeseen problems in the general case

File: src/pkg/services/m365/api/events.go [hunk 2]
```diff
@@ -367,6 +367,59 @@ func (c Events) DeleteItem(
...
+		Users().
+		ByUserId(userID).
+		Calendars().
+		ByCalendarId(containerID).
+		Events().
+		ByEventId(itemID).
+		Attachments().
+		Post(ctx, body, nil)
+	if err != nil {
+		return graph.Wrap(ctx, err, "uploading small mail attachment")
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/events.go:390]:
```suggestion
		return graph.Wrap(ctx, err, "uploading small event attachment")
```

File: src/pkg/services/m365/api/events.go [hunk 3]
```diff
@@ -367,6 +367,59 @@ func (c Events) DeleteItem(
...
+		ByUserId(userID).
+		Calendars().
+		ByCalendarId(containerID).
+		Events().
+		ByEventId(itemID).
+		Attachments().
+		CreateUploadSession().
+		Post(ctx, session, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "uploading large mail attachment")
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/events.go:417]:
```suggestion
		return nil, graph.Wrap(ctx, err, "uploading large event attachment")
```

**Review by ashmrtn — APPROVED** [review]

File: src/internal/connector/exchange/restore_test.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/restore_test.go]: this will change significantly in an upcoming PR, and will reduce the duplicate-value population.

**Pull request merged**